sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs

```diff
@@ -177,14 +179,14 @@ public SharedAccessSignature(string sharedAccessSignature) : this(sharedAccessSi
         }
 
         /// <summary>
-        ///   Extends the period for which the shared access signature is considered valid by adjusting the
-        ///   calculated expiration time.  Upon successful extension, the <see cref="Value" /> of the signature will
-        ///   be updated with the new expiration.
+        ///   Creates a new signature with an extended period for which the shared access signature is considered valid.
         /// </summary>
         ///
         /// <param name="signatureValidityDuration">The duration that the signature should be considered valid.</param>
         ///
-        public void ExtendExpiration(TimeSpan signatureValidityDuration)
+        /// <returns>A new <see cref="SharedAccessSignature" /> based on the same key, but with an extended expiration time.</returns>
+        ///
+        public SharedAccessSignature CloneAndExtendExpiration(TimeSpan signatureValidityDuration)
```

Comment:
I know that's how it's intended to be used, but I think the name does not fully correspond to the method behavior. We could specify a duration that's shorter than the one we have now. We could:
- Assert the new duration is greater than the one we had previously.
- Use the argument to actually extend the time (use what we have in the original instance + the specified value).
- Change the method name to something like `CloneAndChangeExpiration`, `CloneAndUpdateExpiration`, `CloneAndRedefineExpiration`.

Reply:
I see your point.   I think the rename without the restrictions is the better way to go.   This is an internal only class, so I want to keep it less rigid.